packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -883,16 +883,16 @@ export class Bucket extends BucketBase {
       ? false
       : attrs.bucketWebsiteNewUrlFormat;
 
-    const websiteUrl = newUrlFormat
+    const websiteDomain = newUrlFormat
       ? `${bucketName}.s3-website.${region}.${urlSuffix}`
       : `${bucketName}.s3-website-${region}.${urlSuffix}`;
 
     class Import extends BucketBase {
       public readonly bucketName = bucketName!;
       public readonly bucketArn = parseBucketArn(scope, attrs);
       public readonly bucketDomainName = attrs.bucketDomainName || `${bucketName}.s3.${urlSuffix}`;
-      public readonly bucketWebsiteUrl = attrs.bucketWebsiteUrl || websiteUrl;
-      public readonly bucketWebsiteDomainName = Fn.select(2, Fn.split('/', attrs.bucketWebsiteUrl || websiteUrl));
+      public readonly bucketWebsiteUrl = attrs.bucketWebsiteUrl || `http://${websiteDomain}`;
+      public readonly bucketWebsiteDomainName = attrs.bucketWebsiteUrl ? attrs.bucketWebsiteUrl.split('/')[2] : websiteDomain;
```

Comment:
Since you don't know if `attrs.bucketWebsiteUrl` is a token or a concrete value, use `Fn.split` and `Fn.select` to parse it. Otherwise, it will fail if `bucketWebsiteUrl` is a token.